src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -156,6 +245,113 @@ public abstract partial class JsonNamingPolicy
         public static System.Text.Json.JsonNamingPolicy CamelCase { get { throw null; } }
         public abstract string ConvertName(string name);
     }
+    public partial class JsonNode
+    {
+        public JsonNode() { }
+    }
+    public partial class JsonNull : System.Text.Json.JsonNode
+    {
+        public JsonNull() { }
+    }
+    public partial class JsonNumber : System.Text.Json.JsonNode
+    {
+        public JsonNumber() { }
+        public JsonNumber(bool value) { }
+        public JsonNumber(byte value) { }
+        public JsonNumber(double value) { }
+        public JsonNumber(short value) { }
+        public JsonNumber(int value) { }
+        public JsonNumber(long value) { }
+        [System.CLSCompliantAttribute(false)]
+        public JsonNumber(sbyte value) { }
+        public JsonNumber(float value) { }
+        public JsonNumber(string value) { }
```

Comment:
I would remove this as well, and let the user have to do int.parse(string) instead.

Reply:
This would be useful for writing down numbers not representable in the numeric types.  The canonical test in JsonDocument/JsonElement is `1e400`, which is Infinity in double.

If the payload contains 1e400 the JsonNode tree needs to do something, at which point a user should maybe be able to do it, too.

Reply:
Right, thanks for pointing this out, I'll keep it then.